lib/countryParser.py

```diff
@@ -0,0 +1,49 @@
+from lxml import etree
+
+from helpers.logHelpers import createLog
+from helpers.errorHelpers import ProcessingError
+
+logger = createLog('country_code_generator')
+
+
+def loadCountryCodes():
+    """HathiTrust records include publication places encoded in the MARC
+    country code format. These codes are accessed via an XML file from LOC that
+    provides the full-text versions of these country names. This function
+    generates a dict that correlates these codes with the full-text values,
+    which is used to translate the codes for use in SFR
+    """
+
+    countryTrans = {}
+
+    logger.info('Parsing country code/names dict from local file')
+
+    try:
+        countryXML = open('lib/marc_countries.xml')
+    except FileNotFoundError:
+        logger.error('Could find file at path {}'.format(
+            'lib/marc_countries.xml'
+        ))
+        raise ProcessingError(
+            'loadCountryCodes',
+            'Could not open marc_countries.xml'
+        )
+
+    with countryXML as countryXML:
+        countryTree = etree.parse(countryXML)
+        codeList = countryTree.getroot()
+        countries = codeList.find('{info:lc/xmlns/codelist-v1}countries')
```

Comment:
Just a style thing, but more pythonic would be to construct the `dict` out of a list comprehension:

    countryTrans = dict([(x.find('{info:lc/xmlns/codelist-v1}code').text, x.find('{info:lc/xmlns/codelist-v1}name').text) for x in codeList.findall('.//{info:lc/xmlns/codelist-v1}country')])

Reply:
Very true, looking at this again I don't know why I was looping to get the child elements of each country, so inefficient!